app/controllers/security_group_controller.rb

```diff
@@ -14,4 +14,247 @@ def self.display_methods
   end
 
   menu_section :net
+
+  def button
+    @edit = session[:edit] # Restore @edit for adv search box
+    params[:display] = @display if %w(vms instances images).include?(@display)
+    params[:page] = @current_page unless @current_page.nil? # Save current page for list refresh
+
+    @refresh_div = "main_div"
+    return tag("SecurityGroup") if params[:pressed] == "security_group_tag"
+    delete_security_groups if params[:pressed] == 'security_group_delete'
+
+    if params[:pressed] == "security_group_edit"
+      checked_security_group_id = get_checked_security_group_id(params)
+      javascript_redirect :action => "edit", :id => checked_security_group_id
+    elsif params[:pressed] == "security_group_new"
+      javascript_redirect :action => "new"
+    elsif !flash_errors? && @refresh_div == "main_div" && @lastaction == "show_list"
+      replace_gtl_main_div
+    else
+      render_flash
+    end
+  end
```

Comment:
we really need to make the "button" and "x_button" actions simple like here: https://github.com/ManageIQ/guides/blob/master/ui/button_actions.md

or at least make it a simple `case` so that it can be easily refactored later.

Someone is going to have a hell of a job refactoring the remaining `button` methods so let's  not make it worse with new code.

Btw similar code:
```
app/controllers/cloud_network_controller.rb:24
app/controllers/cloud_subnet_controller.rb:16
app/controllers/network_router_controller.rb:16
```